src/utilities/extendSchema.js

```diff
@@ -150,8 +151,30 @@ export function extendSchema(
         }
         directiveDefinitions.push(def);
         break;
-      case Kind.SCALAR_TYPE_EXTENSION:
       case Kind.INTERFACE_TYPE_EXTENSION:
+        const extendedInterfaceTypeName = def.name.value;
+        const existingInterfaceType = schema.getType(extendedInterfaceTypeName);
+        if (!existingInterfaceType) {
+          throw new GraphQLError(
+            `Cannot extend interface "${extendedInterfaceTypeName}" because ` +
+              'it does not exist in the existing schema.',
+            [def],
+          );
+        }
+        if (!isInterfaceType(existingInterfaceType)) {
+          throw new GraphQLError(
+            `Cannot extend non-interface type "${extendedInterfaceTypeName}".`,
+            [def],
+          );
+        }
+        typeExtensionsMap[
+          extendedInterfaceTypeName
+        ] = appendExtensionToTypeExtensions(
+          def,
+          typeExtensionsMap[extendedInterfaceTypeName],
+        );
```

Comment:
You need to test multiple extension of the same interface:
https://github.com/graphql/graphql-js/blob/b333b30b238d040d6a8cf959d54f037ee26ae2c6/src/utilities/__tests__/extendSchema-test.js#L732